internal/interceptor.go

```diff
@@ -299,6 +299,22 @@ type ClientOutboundInterceptor interface {
 	mustEmbedClientOutboundInterceptorBase()
 }
 
+// ScheduleClientInterceptor provides a way to intercept
+// certain schedule-specific client calls from the SDK. See documentation in the
+// interceptor package for more details.
+type ScheduleClientInterceptor interface {
```

Comment:
If you must have a separate interceptor, you should rename this to `ScheduleClientOutboundInterceptor` and add `InterceptScheduleClient(next ScheduleClientOutboundInterceptor) ScheduleClientOutboundInterceptor` to `ClientInterceptor` and invoke it with your underlying interceptor on instantiation. I would also support just putting `CreateSchedule` on the `ClientOutboundInterceptor`.